td/tests/tracking/test_views.py

```diff
@@ -0,0 +1,966 @@
+        # SessionWizardView will try to use session. Setting session up here.
+        engine = import_module(settings.SESSION_ENGINE)
+        store = engine.SessionStore()
+        store.save()
+        setattr(self.request, "session", store)
+        # Mocking messages framework so the view can use it
+        setattr(self.request, "_messages", FallbackStorage(self.request))
+        self.view = setup_view(MultiCharterEventView(), self.request)
+
+    def test_get(self):
+        """
+        Sanity check against errors when going to multi-charter event form
+        """
+        response = MultiCharterEventView.as_view()(self.request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_config(self):
+        """
+        Sanity check for config
+        """
+        self.assertEqual(self.view.template_name, "tracking/multi_charter_event_form.html")
+        self.assertEqual(self.view.form_list, [MultiCharterStarter, MultiCharterEventForm2])
+        self.assertEqual(self.view.initial_dict, {"1": {"start_date": timezone.now().date()}})
+
+    @patch("td.tracking.views.MultiCharterEventForm2")
+    def test_get_context_data(self, mock_form_2):
+        post_data = {
+            "translator0": "Test Translator",
+            "facilitator0": "Test Facilitator",
+            "is_lead0": True,
+            "speaks_gl0": True,
+            "material0": "Test Material",
+            "licensed0": True,
+        }
+        self.request = RequestFactory().post('/tracking/mc-event/new/', post_data)
+        self.view = setup_view(MultiCharterEventView(), self.request)
+        # For some reason, super's get_context_data() will look for the following attributes
+        self.view.storage = lambda: None
+        setattr(self.view.storage, "extra_data", {})
+        self.view.steps = lambda: None
+        setattr(self.view.steps, "current", "1")
+        self.view.prefix = "multi_charter_event_form"
+        #
+        context = self.view.get_context_data(mock_form_2)
+        #
+        self.assertEqual(context["translators"][0], {"name": "Test Translator"})
+        self.assertEqual(context["facilitators"][0], {"name": "Test Facilitator", "is_lead": True, "speaks_gl": True})
+        self.assertEqual(context["materials"][0], {"name": "Test Material", "licensed": True})
+
+    @patch("td.tracking.views.Language.objects.get")
+    def test_get_form(self, mock_get):
+        """
+        In step 0 without POST, the form returned should be MultiCharterStarter
+        In step 0 with POST, the form returned should MultiCharterEventForm1
+        In step 1, the form returned should be MultiCharterEventForm2
+        """
+        self.assertIsInstance(self.view.get_form(step=0), MultiCharterStarter)
+
+        # instance_dict will be accessed by an internal code
+        self.view.instance_dict = {"1": Event}
+        self.assertIsInstance(self.view.get_form(step=1), MultiCharterEventForm2)
+
+        language = lambda: None

```

Comment:
I believe this would also work and might be a bit more clear that you are mocking a language:

```
language = Mock(
    id="9999",
    ln="Test Language",
    lc="tl",
   etc
)

mock_get.return_value = language
```